tools/online_editor/src/preview_widget.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+// cSpell: ignore lumino
+
+import { Widget } from "@lumino/widgets";
+
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+
+import slint_init, * as slint from "@preview/slint_wasm_interpreter.js";
+
+async function setup_preview() {
+  await slint_init();
+}
+
+export class PreviewWidget extends Widget {
+  private instance: slint.WrappedInstance | null;
+  private canvas_id: string;
+
+  static createNode(): HTMLElement {
+    const node = document.createElement("div");
+    const content = document.createElement("div");
+    content.className = "preview-container";
+
+    const canvas = document.createElement("canvas");
+    content.appendChild(canvas);
+
+    const error_area = document.createElement("div");
+    error_area.className = "error-area";
+    content.appendChild(error_area);
+
+    node.appendChild(content);
+
+    return node;
+  }
+
+  constructor() {
+    super({ node: PreviewWidget.createNode() });
+    this.setFlag(Widget.Flag.DisallowLayout);
+    this.addClass("content");
+    this.addClass("preview");
+    this.title.label = "Preview";
+    this.title.caption = `Slint code preview`;
```

Comment:
"Preview" is a little bit of a misnomer here, I think. This is IMO the equivalent of `slint-viewer` for the web.
```suggestion
    this.title.caption = `Slint Viewer`;
```